src/algorithms/excitation/TM_excitations.jl

```diff
@@ -0,0 +1,26 @@
+function correlation_length(state::InfiniteMPS; otherstate=state, tol = 1e-8, exi_space = ComplexSpace(1), tol_angle = 0.1, triv=true)
+    ALs1 = state.AL
+    ALs2 = otherstate.AL
+
+   #the first eigenvalue tells us what angle to search at for the leading nontrivial eigenvalue
+    init = TensorMap(rand, ComplexF64, space(ALs2[1],1)*exi_space,space(ALs1[1],1))
+
+    eigenvals, _ = eigsolve(init, 20, :LM, tol=tol) do x
```

Comment:
this can be problematic, when the transfer matrix has dimension < 20. I would also maybe move the 20 to the keyword arguments?

Reply:
good point. This is a bit arbitrary we just need to ensure that the second singular value in the requested symmetry sector is in there.